src/components/PupilViews/PupilReview/PupilReview.view.tsx

```diff
@@ -73,6 +73,45 @@ export const PupilViewsReview = (props: PupilViewsReviewProps) => {
   const [isAttemptingShare, setIsAttemptingShare] = useState<
     "failed" | "shared" | "initial"
   >("initial");
+  const [storedAttemptLocally, setStoredAttemptLocally] = useState<{
+    stored: boolean;
+    attemptId: string;
+  }>({ stored: false, attemptId: "" });
+
+  useEffect(() => {
```

Comment:
you don't need use effect for this and you shouldn't use it.  There is a  risk of side effects and recursive render loops.  You can just put this in the main render and check that storedAttemptLocally.stored is false before calling.